### ISSUE TITLE
error in function phyloseq_filter_prevalence

### ISSUE THREAD

**Issue by J5886** [vmikk/metagMisc#28]
Hello,

I'm using the function phyloseq_filter_prevalence to filter a phyloseq object with R version 4.2.3 and it's working correctly with me, when I upgraded R version to v. 4.3.1 I got an error "OTU abundance data must have non-zero dimensions" with the same data,
R version was switched to the old version 4.2.3 to check again and it worked
the command:

physeq3 <- phyloseq_filter_prevalence(physeq.gen, prev.trh = 0.5, abund.trh = 10, threshold_condition = "OR")
Error in validObject(.Object) : invalid class “otu_table” object:
OTU abundance data must have non-zero dimensions.

 thanks for your help in advance

**Comment by vmikk** [comment]
Hello!

Thank you for reporting the issue.
However, I could not reproduce it. E.g., the function works for the `GlobalPatterns` data:
``` R
library(phyloseq)
library(metagMisc)
data(GlobalPatterns)
pf <- phyloseq_filter_prevalence(GlobalPatterns, prev.trh = 0.5, abund.trh = 10, threshold_condition = "OR")
pf
```

I think that the filtering thresholds you set are too strict for your data, which results in the removal of all OTUs/ASVs (there are no OTUs that occur in 50% of the samples or have an abundance greater than 10 reads). The message `OTU abundance data must have non-zero dimensions` means that the `otu_table` function from `phyloseq` can not construct the object without data.

Although, probably I need to add some meaningful error messages.

HTH,
With kind regards,
Vladimir

**Comment by J5886** [comment]
Hi @vmikk 
thanks for your reply
I got it, but even when the threshold was reduced to 10% "prev.trh = 0.1" and the abundance to 5 reads, I got the same error, with the old version of R I used various thresholds and it worked smoothly.
I checked that with other data and found the error again.

Best regards

**Comment by hkiesewalter** [comment]
Hi, I have the exact same issue as described by J5886 with my own data.
That error does not appear when using the GlobalPatterns data, but the number of remaining taxa after filtration is very low and not the same as in your example.

![Screenshot 2023-11-18 at 19 05 14](https://github.com/vmikk/metagMisc/assets/85175678/25907d1e-11d2-4417-ab2b-5b8171d1a22d)

**Comment by hkiesewalter** [comment]
Hi all, 

I installed the old version of metagMisc (0.0.4) manually under the newest R version (4.3.2), and the function [phyloseq_filter_prevalence] works totally fine. 

So, I guess the error is somehow in the new version of metagMisc (0.5.0)?

**Comment by JH966** [comment]
how to install 0.0.4 version?

**Comment by vmikk** [comment]
@J5886, @tlmontgo, @hkiesewalter, and @JH966,
Thank you for bringing this issue to my attention, and I apologize for the delay in responding. 

To effectively diagnose and resolve the problem, I need to replicate the issue on my end.
Could one of you kindly provide a small subset of data where the function fails?
This will help me to understand what causes the issue and to work towards a solution.
Please ensure that any sensitive information is removed or anonymized before sharing.

**Comment by JH966** [comment]
> Hi all,
> 
> I installed the old version of metagMisc (0.0.4) manually under the newest R version (4.3.2), and the function [phyloseq_filter_prevalence] works fine.
> 
> So, I guess the error is somehow in the new version of metagMisc (0.5.0)?

hi, how to install old version? please .. tell me

**Comment by vmikk** [comment]
To install the older version, you may use

```{R}
devtools::install_github("vmikk/metagMisc@v.0.0.4")   # v.0.0.4
```
or
```{R}
remotes::install_github("vmikk/metagMisc@v.0.0.4")
```

**Comment by hkiesewalter** [comment]
> @J5886, @tlmontgo, @hkiesewalter, and @JH966, Thank you for bringing this issue to my attention, and I apologize for the delay in responding.
> 
> To effectively diagnose and resolve the problem, I need to replicate the issue on my end. Could one of you kindly provide a small subset of data where the function fails? This will help me to understand what causes the issue and to work towards a solution. Please ensure that any sensitive information is removed or anonymized before sharing.

Hi,

I have attached a dataset plus R script, which is causing the error in my data and giving wrong filtration results on the GlobalPattern dataset.

Best regards

[R_metagMisc.zip](https://github.com/vmikk/metagMisc/files/13949141/R_metagMisc.zip)

**Comment by vmikk** [comment]
@hkiesewalter, thank you for sharing the dataset and R script. It was very helpful for pinpointing the problem!

I've identified and fixed the issue (closed with commit 12a8b2e).  
It was related to the recent update of the `prevalence` function, which now utilizes the `data.table` package for enhanced performance. However, `data.table` does not use row names, which is where OTU IDs were previously stored. As a result, subsetting the phylonext object was not functioning as intended.

Please pull the latest commit of the package to incorporate these fixes.
```
remotes::install_github("vmikk/metagMisc")
```

**Comment by JH966** [comment]
> @hkiesewalter, thank you for sharing the dataset and R script. It was very helpful for pinpointing the problem!
> 
> I've identified and fixed the issue (closed with commit [12a8b2e](https://github.com/vmikk/metagMisc/commit/12a8b2e1c3107ee63abbc8c2884dbc681c24d421)). It was related to the recent update of the `prevalence` function, which now utilizes the `data.table` package for enhanced performance. However, `data.table` does not use row names, which is where OTU IDs were previously stored. As a result, subsetting the phylonext object was not functioning as intended.
> 
> Please pull the latest commit of the package to incorporate these fixes.
> 
> ```
> remotes::install_github("vmikk/metagMisc")
> ```

thank you!!!!! very very thank you!!!

**Comment by hkiesewalter** [comment]
@vmikk Thanks a lot for fixing it so quickly 👍 :)
